daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -2177,7 +2180,42 @@ case class DFDLXShiftExpr(nameAsParsed: String, fnQName: RefQName,
     res
   }
 }
+case class DFDLXBitExpr(nameAsParsed: String, fnQName: RefQName,
+                        args: List[Expression], constructor: (List[CompiledDPath], NodeInfo.Kind,NodeInfo.Kind) => RecipeOp)
+  extends FunctionCallBase(nameAsParsed, fnQName, args) {
+  override lazy val inherentType = {
+    val argInherentType = args(0).inherentType
+    val arg2=args(1).inherentType
+    schemaDefinitionUnless((
+      argInherentType.isSubtypeOf(NodeInfo.PrimType.UnsignedLong) && arg2.isSubtypeOf(NodeInfo.PrimType.UnsignedLong))||  (argInherentType.isSubtypeOf(NodeInfo.PrimType.Long) && arg2.isSubtypeOf(NodeInfo.PrimType.Long)),
+      "First argument and Second argument both should  be either xs:unsignedLong  or a subtype of those or xs:long or a subtype of those",
+      nameAsParsed,argInherentType.globalQName)
+    argInherentType
+  }
+
 
+  override def targetTypeForSubexpression(subexp: Expression): NodeInfo.Kind = {
+    if (subexp == args(0))
+      inherentType
+    else if (subexp == args(1))
+      inherentType
+    else
+    //$COVERAGE-OFF$
+      Assert.invariantFailed("subexpression %s is not an argument.".format(subexp))
+    //$COVERAGE-ON$
+  }
```

Comment:
I think you can just have
```
override def targetTypeForSubexpression(subexp: Expression): NodeInfo.Kind = inherentType
```
because all the logic to reconcile the two arg types is over in the inherentType lazy val. 

I think the check for the subexp somehow not being one of the arguments - you don't need to check for that. It's an invariant check anyway, and really shouldn't be the responsibility of every operator like this one to check for that. (If that invariant is worth checking, the caller of targetTypeForSubexpression should do that check.)